source/migrating_from_legacy_products/index.html.md.erb

```diff
@@ -3,6 +3,121 @@ title: Migrate from our old frameworks
+
+You can use your list to see where you can replace similar elements in your service with one GOV.UK Frontend component. For example, if you have a range of similar buttons, you may be able to replace them all with the [button](https://design-system.service.gov.uk/components/button/) component.
+
+We recommend using a template engine like [Nunjucks](https://mozilla.github.io/nunjucks/) to abstract each component into a reusable macro (also called a 'partial' or 'template'). You can do this before or during your migration.
+
+If you use macros, your project will be easier to:
+
+- migrate, because your project will be organised the same way we organise the GOV.UK Design System
+- update when new versions of GOV.UK Frontend are released
+
+If you're using Node.js, you can use [our Nunjucks macros](https://design-system.service.gov.uk/styles/page-template/#nunjucks). You'll need to [install Nunjucks](https://mozilla.github.io/nunjucks/getting-started.html) first.
+
+## Include GOV.UK Frontend in your project
+
+1. [Install GOV.UK Frontend using node package manager (npm)](/installing_with_npm).
+
+    Do not install from `dist`, or you will not be able to turn on 'compatibility mode'.
+
+2. [Turn on 'compatibility mode'](/compatibility_mode/#turn-on-39-compatibility-mode-39) so GOV.UK Frontend uses the colours and font from our old frameworks.
+
+3. Test your application to make sure it looks the same and behaves correctly.
+
+## Migrate your pages and components
+
+After you install GOV.UK Frontend, you can either migrate:
+
+- one page at a time - if your service is small, for example a single service with 10 pages
+- one component at a time - if your service is big or complicated
+
+### Migrate one page at a time
+
+We recommend creating a separate page layout based on our [page template](https://design-system.service.gov.uk/styles/page-template/), so that you can make changes without affecting the pages you haven't migrated yet.
+
+You must:
+
+- use GOV.UK Frontend markup for all the [typography](https://design-system.service.gov.uk/styles/typography/) and [layout](https://design-system.service.gov.uk/styles/layout/) in your service
+- include and initialise GOV.UK Frontend's JavaScript for all the components that use it
+
+Read about [how GOV.UK Pay migrated their service one page at a time](https://technology.blog.gov.uk/2018/12/21/the-benefits-of-migrating-gov-uk-pays-codebase-to-the-gov-uk-design-system/).
+
+### Migrate one component at a time
+
+We recommend migrating in the following order:
+
+- [components](https://design-system.service.gov.uk/components/)
+- [typography](https://design-system.service.gov.uk/styles/typography/)
+- [header](https://design-system.service.gov.uk/components/header/)
+- [footer](https://design-system.service.gov.uk/components/footer/)
+- [page template](https://design-system.service.gov.uk/styles/page-template/)
```

Comment:
Agree on the first point - but (as I recently found out when I tried to follow the Frontend install instructions) the `js-enabled` requirement isn't necessarily clear to anyone using Frontend, regardless of whether they're migrating. I think it's something we're going to cover more explicitly in the install guide - so that should mean we won't need to mention it specifically here.

Reply:
On the second point - do you think it's ok not to mention this at this stage, because the work on getting everything looking exactly right doesn't come until later (under '## Remove our old frameworks')?

Reply:
That's a great point about the `js-enabled` class, though we do implictly rely on the fact that GOV.UK Template adds the `js-enabled` class as well – Mark actually ran into this when trying to do the user research task with the accordion component.

In terms of the viewport meta tag, one thing we should be conscious of is that we've changed it to add `viewport-fit=cover` to support devices with notches. We probably **don't** want that added until they've finished migrating, as GOV.UK Template and Elements do not do the necessary margin calculations.